gke-gs-bucket-backup/gs_bucket_sync.py

```diff
@@ -120,49 +120,55 @@ def tar_directory(self, tar_dir: str) -> str:
     def upload_encrypted_timestamp_file(self, upload_file: str) -> None:
         """ Given a file path upload said file to our backup bucket.
             File is timestamp'd and includes file prefix. """
-        timestamp_name = datetime.datetime.now().strftime(
-            '%F-%R-%s-{}.tar.gz'.format(self.suffix))
-        backup_bucket_path = os.path.join(self.dst, timestamp_name)
+        timestamp = datetime.datetime.now().strftime('%F-%R-%s')
+        backup_bucket_path = os.path.join(self.dst, '{}-{}'.format(timestamp, self.filename))
 
         self.gsutil_encrypt_cp_cmd(upload_file, backup_bucket_path)
 
 
 if __name__ == "__main__":
     args = argparse.ArgumentParser(description=__doc__)
-    args.add_argument('src', type=str, help="Source bucket")
-    args.add_argument('backup', type=str, help="Backup bucket")
-    args.add_argument(
-        'suffix', type=str,
-        help="Backup suffix filename ex: $timestamp-prefix.tar.gz")
-    args.add_argument('-v', action='store_true', default=False,
+    args.add_argument('-v', '--verbose', action='store_true', default=False,
                       required=False, help='Increase verbosity')
-    args.add_argument('-gsutil_bin', type=str, required=False,
+    args.add_argument('-g', '--gsutil', type=str, required=False,
                       help="Full path to gsutil binary on disk",
                       default="/usr/bin/gsutil")
-    args.add_argument('-svc_act_key', type=str, required=False,
+    args.add_argument('-s', '--svc-acct-key', type=str, required=False,
                       help="Full path to a GCP service account key")
     parsed_args = args.parse_args()
 
     # Set verbose output
-    if parsed_args.v:
+    if parsed_args.verbose:
         logger.setLevel(logging.DEBUG)
 
     logger.debug("ARGS piped in: {}".format(parsed_args))
 
-    try:
-        encrypt_key = os.environ['GBACKUP_ENV_KEY']
-    except KeyError:
-        logging.error("Must expose an env var called GBACKUP_ENV_KEY")
+    backup_src = os.environ.get("GS_BACKUP_SRC")
+    backup_dest = os.environ.get("GS_BACKUP_DEST")
+    filename = os.environ.get("GS_BACKUP_FILENAME")
+    encrypt_key = os.environ.get("GS_ENCRYPTION_KEY")
```

Comment:
Works well enough! You could fold all of these lookups into the `parser` logic, with `default=os.environ.get("FOO")` for each one, which might make the parsing a bit more intuitive to other maintainers.

Reply:
Okay, you nerdsniped me -- the argument parsing is now much more verbose with options overriding the environ defaults (Including one for a path to a file containing the encryption key, since that's not safe to put on the command line).